doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,279 @@
+# Solve an equation algebraically
+
+Use SymPy to solve an equation algebraically (symbolically). For example, solving $x^2 = y$ yields $x \in \{-\sqrt{y},\sqrt{y}\}$.
+
+There are two high-level functions to solve equations, {func}`~.solve` and {func}`~.solveset`. Here are recommendations on when to use:
+
+- {func}`~.solve`
+    - You need to programmatically extract components (expressions, or individual symbols or constants) from the output by traversing it.
+    - You want an explicit solution expression that you can use to substitute into something else.
+
+- {func}`~.solveset`
+    - You want a consistent input and output interface.
+    - You want to get all the solutions, including if there are infinitely many.
+    - You want a clear separation between equations in the complex domain and the real domain.
+
+We recommend you include the variable to be solved for as the second argument for either function. While this is optional for equations with a single symbol, it is a good practice because it ensures SymPy will solve for the desired symbol. For example, you may expect the following to solve for $x$, and SymPy will solve for $y$:
+
+```py
+>>> from sympy.abc import x, y
+>>> from sympy import solve
+>>> solve(x ** 2 - y)
+[{y: x**2}]
+```
+
+Specifying the variable to solve for ensures that SymPy solves for it:
+
+```py
+>>> from sympy.abc import x, y
+>>> from sympy import solve
+>>> solve(x ** 2 - y, x)
+[-sqrt(y), sqrt(y)]
+```
+
+## Using {func}`~.solve`
+
+- produces various output formats depending on the answer
+    - unless you use `dict=True` to ensure the result will be formatted as a dictionary, which we recommend if you want to extract information from the result programmatically
+
+You can solve an equation using {func}`~.solve` in several ways.
+
+### Make your equation into an expression that equals zero
+
+Use the fact that any expression not in an `Eq` (equation) is automatically assumed to equal zero (0) by the solving functions. You can rearrange the equation $x^2 = y$ to $x^2 - y = 0$, and {func}`~.solve` that expression. This approach is convenient if you are interactively solving an equation which can easily be rearranged to $expression = 0$.
```

Comment:
It's pretty trivial to turn an equation into an expression:
```python
In [9]: eq = Eq(x, y)

In [10]: eq.lhs - eq.rhs
Out[10]: x - y

In [11]: eq.rewrite(Add)
Out[11]: x - y
```
The reason for not doing this is if it would cancel a singularity:
```python
In [12]: eq = Eq(x + 1/x, 1/x)

In [13]: eq
Out[13]: 
    1   1
x + ─ = ─
    x   x

In [14]: eq.rewrite(Add)
Out[14]: x

In [15]: eq.rewrite(Add, evaluate=False)
Out[15]: 
    1   1
x - ─ + ─
    x   x
```
I think many users expect that the input to `solve` should be equations rather than just expressions. This point about `Eq` should be made right at the beginning perhaps the first example should be with an equation.

Another terminology for understanding what it means to set an expression equal to zero is that we want to "find roots" of functions.

Reply:
The singularity thing seems like a very technical difference that probably isn't worth mentioning here.

Reply:
For most users, it comes down to whether they more naturally think of their problem as an equality or as an expression that equals 0. Also, from personal experience, I often use Eq just because it's easier to type than negating the entire right-hand side of an expression, and also less error prone (I might mess up minus sign with `-rhs`).

Reply:
It does but from experience I can say that a lot of people find the idea of thinking of an expression as an equation implicitly set to zero unintuitive so this is something to be discussed early on.

Also the example should output what the Eq looks like so that it's clear(er) what a symbolic equality is.